featuremanagement/azuremonitor/_send_telemetry.py

```diff
@@ -51,17 +53,49 @@ def publish_telemetry(evaluation_event: EvaluationEvent) -> None:
     """
     if not HAS_AZURE_MONITOR_EVENTS_EXTENSION:
         return
-    event = {}
-    if evaluation_event.feature:
-        event[FEATURE_NAME] = evaluation_event.feature.name
+    event: Dict[str, Optional[str]] = {}
+    if not evaluation_event.feature:
+        return
+    event[FEATURE_NAME] = evaluation_event.feature.name
     event[ENABLED] = str(evaluation_event.enabled)
+    event["Version"] = EVALUATION_EVENT_VERSION
 
+    # VariantAllocationPercentage
     if evaluation_event.reason and evaluation_event.reason != VariantAssignmentReason.NONE:
         if evaluation_event.variant:
             event[VARIANT] = evaluation_event.variant.name
         event[REASON] = evaluation_event.reason.value
 
-    if evaluation_event.feature and evaluation_event.feature.telemetry:
+        if evaluation_event.reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED:
+            allocation_percentage = 0
+
+            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
+                for allocation in evaluation_event.feature.allocation.percentile:
+                    if (
+                        evaluation_event.variant
+                        and allocation.variant == evaluation_event.variant.name
+                        and allocation.percentile_to
+                    ):
+                        allocation_percentage += allocation.percentile_to - allocation.percentile_from
+
+            event["VariantAssignmentPercentage"] = str(100 - allocation_percentage)
```

Comment:
Is the reason of this type conversion (number to string) that we want to put it in the customDimension instead of customMetrics of the application insights event?

Reply:
The type of `event` is `Dict[str, Optional[str]]`